functions/system.bash

```diff
@@ -40,36 +35,33 @@ needed_packages() {
   # Install avahi-daemon - hostname based discovery on local networks
   # Install python/python-pip - for python packages
   echo -n "$(timestamp) [openHABian] Installing additional needed packages... "
-  #cond_redirect apt-get update
-  cond_redirect apt-get -y install apt-transport-https bc sysstat avahi-daemon python python-pip avahi-autoipd fontconfig
-  if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; exit 1; fi
+  if cond_redirect apt-get -y install apt-transport-https bc sysstat avahi-daemon python python-pip avahi-autoipd fontconfig; then echo "OK"; else echo "FAILED"; exit 1; fi
 
   if is_pithree || is_pithreeplus || is_pizerow; then
     echo -n "$(timestamp) [openHABian] Installing additional bluetooth packages... "
-    cond_redirect apt-get -y install bluez python-bluez python-dev libbluetooth-dev raspberrypi-sys-mods pi-bluetooth
-    if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; exit 1; fi
+    if cond_redirect apt-get -y install bluez python-bluez python-dev libbluetooth-dev raspberrypi-sys-mods pi-bluetooth; then echo "OK"; else echo "FAILED"; exit 1; fi
   fi
 }
 
 timezone_setting() {
+  # shellcheck source=/etc/openhabian.conf disable=SC1091
   source "$CONFIGFILE"
   if [ -n "$INTERACTIVE" ]; then
     echo -n "$(timestamp) [openHABian] Setting timezone based on user choice... "
     dpkg-reconfigure tzdata
-  elif [ ! -z "${timezone+x}" ]; then
+  elif [ -n "${timezone:-Europe/Berlin}" ]; then
```

Comment:
This change will break this routine. Have a look at: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
Insert exception # shellcheck disable=SC2154

Reply:
Sure? I thought it would be functionally a little different but still the right thing.
Anyway I changed it back.

Reply:
Yea, because if the timezone is unset it should fetch one using "geoip" solution. Next "if block"